core/primitives/src/block.rs

```diff
@@ -81,36 +90,14 @@ pub enum Block {
 }
 
 impl Block {
-    pub(crate) fn block_from_protocol_version(
-        this_epoch_protocol_version: ProtocolVersion,
-        header: BlockHeader,
-        body: BlockBody,
-    ) -> Block {
-        if !ProtocolFeature::BlockHeaderV4.enabled(this_epoch_protocol_version) {
-            Block::BlockV2(Arc::new(BlockV2 {
-                header,
-                chunks: body.chunks().to_vec(),
-                challenges: body.challenges().to_vec(),
-                vrf_value: *body.vrf_value(),
-                vrf_proof: *body.vrf_proof(),
-            }))
-        } else if !ProtocolFeature::StatelessValidation.enabled(this_epoch_protocol_version) {
-            // BlockV3 should only have BlockBodyV1
-            match body {
-                BlockBody::V1(body) => Block::BlockV3(Arc::new(BlockV3 { header, body })),
-                _ => {
-                    panic!("Attempted to include newer BlockBody version in old protocol version")
-                }
-            }
-        } else {
-            // BlockV4 and BlockBodyV2 were introduced in the same protocol version `ChunkValidation`
-            // We should not expect BlockV4 to have BlockBodyV1
-            match body {
-                BlockBody::V1(_) => {
-                    panic!("Attempted to include BlockBodyV1 in new protocol version")
-                }
-                _ => Block::BlockV4(Arc::new(BlockV4 { header, body })),
+    pub(crate) fn block_from_protocol_version(header: BlockHeader, body: BlockBody) -> Block {
```

Comment:
super nit: we might want to rename this as we are no longer passing the protocol version? I'm fine to leave this as is as maybe in the future we may need to add the protocol version.

Reply:
Good catch, renamed.